README.md

```diff
@@ -116,44 +115,132 @@ You can additionnaly setup a logger to chose where logs will be output.
     wifiProxy->Start();
 ```
 
-## Architecture
+## Architecture overview
+
+Proxy_wifi is constituted of a service component (proxy_wifi service) meant to
+run on a Windows host, and a driver component (proxy_wifi driver) meant to run
+on a guest virtual machine. This repository contains the implementation of
+proxy_wifi service for Windows. Currently, an implementation of proxy_wifi
+driver exists only for Linux
+
+![General architecture](./doc/proxy_wifi_general_architecture.png)
+
+Proxy_wifi driver role is to intercept Wi-Fi control path related operations
+from the guest and forward them to proxy_wifi service. Communication between
+proxy_wifi driver and proxy_wifi service relies on two hv_sockets: one allows
```

Comment:
Might want to say the unabbreviated term 'Hyper-V sockets (hv_sockets)' here since some people won't know the abbreviation. All following references can then just use the `hv_socket` abbreviation:

```suggestion
proxy_wifi driver and proxy_wifi service relies on two Hyper-V sockets (hvsockets): one allows
```